.changeset/three-shoes-relax.md

```diff
@@ -0,0 +1,61 @@
+---
+'@graphql-eslint/eslint-plugin': major
+---
+
+feat: add new options for `naming-convention` rule
+
+Options for `naming-convention` are changed. New option `types` includes the following kinds:
+
+- `ObjectTypeDefinition`
+- `InterfaceTypeDefinition`
+- `EnumTypeDefinition`
+- `ScalarTypeDefinition`
+- `InputObjectTypeDefinition`
+- `UnionTypeDefinition`
+
+New option `fields` includes the following kinds:
+
+- `FieldDefinition`
+- `InputValueDefinition`
+- `VariableDefinition` (new in v3)
+- `Argument` (new in v3)
+- `DirectiveDefinition` (new in v3)
+
+`QueryDefinition` option is removed in favor of `AST` specific
+selector `FieldDefinition[parent.name.value=Query]`.
+
+Before
+
+```json
+{
+  "@graphql-eslint/naming-convention": [
+    "error",
+    {
+      "ObjectTypeDefinition": "PascalCase",
+      "InterfaceTypeDefinition": "PascalCase",
+      "EnumTypeDefinition": "PascalCase",
+      "ScalarTypeDefinition": "PascalCase",
+      "InputObjectTypeDefinition": "PascalCase",
+      "UnionTypeDefinition": "PascalCase",
+      "FieldDefinition": "camelCase",
+      "InputValueDefinition": "camelCase",
```

Comment:
Is this represented in the `after` set?

Reply:
yes, `fields` option include `InputValueDefinition` :)
![image](https://user-images.githubusercontent.com/7361780/142732051-ca690fd2-5e30-43ec-986f-7f14e0fa5f7b.png)

Reply:
Neat. That could actually make sense. `DirectiveDefinition` seems out of place here though.

Reply:
I took proposition by @ilyavolodin https://github.com/ilyavolodin/graphql-eslint/issues/3#issuecomment-652155524
You are right that it's out of place in terms of the name of the group :)

Reply:
I'm also a little surprised on the application of naming requirements on `Arguments`. If I understand correctly, `Arguments` are key/value pairs provided in operations, the keys of which must match the schema your operation is intended to consume, which you may not have control over. Thus, imposing naming requirements on `Argument` would be imposing only a constraint on which APIs you can consume.

Does that sound accurate? Or am I missing something super obvious 😅 ?

Reply:
What a catch 😍, you are totally right 💯. This was my mistake, I removed `Argument` from `operations-recommended` config and added a relevant test to prevent this mistake in the future 🙏

Reply:
Along that same recommendation from Ilya, I'm curious if it would make sense to remove `fields (boolean)` option from `naming-convention` or add it to `require-description` to start to standardize.

Reply:
I agree to remove that option, because:

1. we will take away the confusing presence of `DirectiveDefinition` in `fields` group.
2. without `DirectiveDefinition` in `fields` group we'll have only 3 kinds in `fields` group, so this group is really small in relation to `types` group where there are 6 kinds

Reply:
That makes most sense to me as well. We already addressed above that naming-convention isn't helpful for `Argument` either, which effectively reduces this to two types, one of which wasn't apparent to me was included under the `fields` naming. I think removing it makes the API easier to use and harder to misuse.

Reply:
`naming-convention` is helpful for `Argument` only in the schema but not in operations. `fields` option was removed 🙂